drivers/counter/counter_ll_stm32_rtc.c

```diff
@@ -225,7 +261,12 @@ static uint32_t rtc_stm32_read(const struct device *dev)
 	ts -= T_TIME_OFFSET;
 
 	__ASSERT(sizeof(time_t) == 8, "unexpected time_t definition");
-	ticks = counter_us_to_ticks(dev, ts * USEC_PER_SEC);
+
+	ticks = ts * counter_get_frequency(dev);
```

Comment:
Since this change impacts non COUNTER_RTC_STM32_SUBSECONDS code, it would be preferable to make it in a dedicated commit with proper justification.
Same for all changes which fall in this category (you can group them per "theme")

Reply:
Done.